src/SonarScanner.MSBuild.Shim/Interfaces/IAdditionalFilesService.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.IO;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarScanner.MSBuild.Shim.Interfaces;
+
+internal interface IAdditionalFilesService
+{
+    /// <summary>
+    /// Searches projectBaseDir for files with extensions that match the languages specified in analysisConfig.
+    /// </summary>
+    IEnumerable<string> AdditionalFiles(AnalysisConfig analysisConfig, DirectoryInfo projectBaseDir);
```

Comment:
FYI, on a follow-up PR I will incorporate the functionality of test detection here, and the contract will change:

```csharp
AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryInfo projectBaseDir);
```

...where `AdditionalFiles` is going to be a class with `Sources` and `Tests` lists as properties.